test/blackbox/run_blackbox_tests/test_cases_key_rotation.c

```diff
@@ -0,0 +1,449 @@
+	char invitation_path_buff[500];
+	struct stat invitation_stat;
+	struct stat temp_stat;
+	struct stat peer_stat;
+	struct stat bar_stat;
+	struct utimbuf timebuf;
+	bool join_status;
+
+	meshlink_destroy("encrypted_conf");
+	meshlink_set_log_cb(NULL, MESHLINK_DEBUG, log_cb);
+
+	// Open a new meshlink instance.
+
+	mesh = meshlink_open_encrypted("encrypted_conf", "foo", "encrypted", DEV_CLASS_BACKBONE, "oldkey", 6);
+	assert_int_not_equal(mesh, NULL);
+
+	// Generate invitations
+
+	char *invitation1 = meshlink_invite(mesh, NULL, "peer");
+	assert_int_not_equal(invitation1, NULL);
+
+	// Read the peer invitation file status structure
+
+	strcpy(invitation_path_buff, "encrypted_conf/invitations/");
+	d = opendir(invitation_path_buff);
+	assert(d);
+
+	if(d) {
+		ent = readdir(d);
+		strcat(invitation_path_buff, ent->d_name);
+		assert(ent);
+		assert(stat(invitation_path_buff, &temp_stat) != -1);
+	}
+
+	closedir(d);
+
+	char *invitation2 = meshlink_invite(mesh, NULL, "bar");
+	assert_int_not_equal(invitation2, NULL);
+
+	// Key rotate the encrypted_conf storage with new key
+
+	bool keyrotation_status = meshlink_encrypted_key_rotate(mesh, "newkey", 6);
+	assert_int_equal(keyrotation_status, true);
+
+	meshlink_close(mesh);
+
+	// Compare invitation file timestamps of old key with new key
+
+	assert(stat(invitation_path_buff, &peer_stat) != -1);
+	assert_int_equal(peer_stat.st_mtime, temp_stat.st_mtime);
+
+	// Change timestamp for @ peer @ node invitation
+
+	timebuf.actime = peer_stat.st_atime;
+	timebuf.modtime = peer_stat.st_mtime - 604805; // > 1 week
+
+	assert(utimes(invitation_path_buff, &timebuf) != -1);
+
+
+	// Reopen the meshlink instance with the new key
+
+	mesh = meshlink_open_encrypted("encrypted_conf", "foo", "encrypted", DEV_CLASS_BACKBONE, "newkey", 6);
+	assert_int_not_equal(mesh, NULL);
+
+	mesh1 = meshlink_open("encrypted_conf.1", "peer", "encrypted", DEV_CLASS_BACKBONE);
+	assert_int_not_equal(mesh1, NULL);
+
+	mesh2 = meshlink_open("encrypted_conf.2", "bar", "encrypted", DEV_CLASS_BACKBONE);
+	assert_int_not_equal(mesh2, NULL);
+
+	assert(meshlink_start(mesh));
+
+	join_status = meshlink_join(mesh1, invitation1);
+	assert_int_equal(join_status, false);
+
+	join_status = meshlink_join(mesh2, invitation1);
+	assert_int_equal(join_status, true);
+
+	// Cleanup
+
+	meshlink_close(mesh);
+	meshlink_close(mesh1);
+	meshlink_close(mesh2);
+	meshlink_destroy("encrypted_conf");
+
+	return true;
+}
+
+/* Execute key rotation Test Case # 5 - Atomicity test */
+static void test_case_key_rotation_05(void **state) {
+	execute_test(test_key_rotation_05, state);
+}
+
+static int break_stage;
+
+static void nop_stage(int stage) {
+	return;
+}
+
+static void debug_probe(int stage) {
+
+	// Terminate the node at the specified stage (by @ break_stage @ )
+	if(stage == break_stage) {
+		raise(SIGINT);
+	} else if((break_stage < 1) || (break_stage > 5)) {
+		fprintf(stderr, "INVALID stage break\n");
+		devtool_keyrotate_probe = nop_stage;
```

Comment:
Maybe you should `raise(SIGBART)` here instead of ignoring the problem? On the other hand, it doesn't make much sense that `break_stage` would be outside the range 1..5, since we control that variable ourselves in this file, so maybe the whole check can go away.